src/terminal/parser/InputStateMachineEngine.cpp

```diff
@@ -814,7 +798,7 @@ DWORD InputStateMachineEngine::_GetCursorKeysModifierState(const gsl::span<const
     //   of the keypad; and the divide (/) and ENTER keys in the keypad.
     // This snippet detects the direction keys + HOME + END
     // actionCode should be one of the above, so just make sure it's not a CSI_F# code
-    if (actionCode < CsiActionCodes::CSI_F1 || actionCode > CsiActionCodes::CSI_F4)
+    if (id < CsiActionCodes::CSI_F1 || id > CsiActionCodes::CSI_F4)

```

Comment:
That depends on the byte order of the VTID to uint64_t conversion but I suppose it will work for now.

Reply:
Yeah, this is a bit icky, but these are all single character codes, so while it's relying on the order of the final characters, that's really no different from the original implementation. I could change it to check each of the four id individually if that's preferable? It should optimize to the same assembly in the end.

Reply:
I'm comfortable with it the way it is. Thanks :smile: